### ISSUE TITLE
Display DataFrame with more than 18 rows/columns

### ISSUE THREAD

**Issue by lungben** [fonsp/Pluto.jl#292]
For larger DataFrames, are the first 18 rows and columns are displayed in Pluto.
Is there a way to increase this number?
I already tried the "standard methods" suggested by DataFrames.jl: setting the COLUMNS and LINES environment variables or using `show`, but without success.

```
using DataFrames

ENV["COLUMNS"]=1000
ENV["LINES"] = 50
# method that works in REPL / IJulia,  but seems to have no effect in Pluto
# https://juliadata.github.io/DataFrames.jl/stable/man/getting_started/#Installation

df = hcat([DataFrame(a=1:30, b=2:31) for _ in 1:10]..., makeunique=true)
# only 18 rows and 18 columns are displayed (the df size is 30x20)

show(df, allrows=true, allcols=true) # prints on the server side, but not in the notebook
```

Could you please help?

**Comment by fonsp** [comment]
That's odd - does IJulia display tables using unicode characters (like the REPL) or with an HTML table?

**Comment by lungben** [comment]
It is printed as HTML table.

![image](https://user-images.githubusercontent.com/52384612/90411000-6367d500-e0ab-11ea-9265-e82a28ba7037.png)

**Comment by fonsp** [comment]
I meant IJulia/Jupyter

**Comment by lungben** [comment]
In IJulia the table is also printed as HTML.
![image](https://user-images.githubusercontent.com/52384612/90506258-3a4f4f00-e154-11ea-9d0f-d1e047176f78.png)

**Comment by Sh4pe** [comment]
I've added the PR #318 (a draft as of now) that would solve the problem. As of now, the environment variables `LINES` and `COLUMNS` would have to be set before importing Pluto however.

**Comment by fonsp** [comment]
Hey so another solution is to use a different IOContext for the `text/html` MIME that for the other MIMEs.

Changing the displaysize of the current IOContext will make html tables bigger, but will also make 2D arrays unreadable (because they use the same variable). But because Pluto has horizontal scrolling for HTML output, we can use a very large value by default - it will probably only affect DataFrames.jl.

The problem then is when you want to see even more rows/columns.

**Comment by lungben** [comment]
Sounds great! 
I am usually setting `ENV["COLUMNS"] = 1000` in my Jupyter notebooks, maybe this is a good starting point.

**Comment by fonsp** [comment]
But isn't 2D array printing also affected?

**Comment by non-Jedi** [comment]
Someone asked about this on zulip, so I thought it would be good to document the solution I offered here as well. You can use the [BrowseTables](https://github.com/tpapp/BrowseTables.jl) package for this:

```julia
using BrowseTables
HTMLTable(df)
```

That will show the whole dataframe (or any other Tables.jl compatible source) as an html table (be careful with very large tables...).

Another solution would be the [TableView](https://github.com/JuliaComputing/TableView.jl) package, but that doesn't currently work because Pluto isn't integrated with [WebIO](https://github.com/JuliaGizmos/WebIO.jl) (#299).

**Comment by lungben** [comment]
BrowseTables does not work very well for me in Pluto because no horizontal scroll bars are shown, thus the output columns are truncated.

In my opinion the user experience would be best if the default display of DataFrames (or any Tables.jl interface) would display all columns (or at least with a very large cut-off, e.g. 100) and displays a scroll-bar if the table gets too broad. 

When displaying a table, it is very rare that I do not want to see all columns (showing only the first e.g. 18 rows is usually fine for me).  The need for an additional package or wrapper command for showing all columns of a DataFrame should be avoided imho.

**Comment by fonsp** [comment]
> The need for an additional package or wrapper command for showing all columns of a DataFrame should be avoided imho.

Agreed!!

In the meantime, I came up with a more stable way to set the IOContext, and I added it to PlutoUI:

```julia
PlutoUI.WithIOContext(df, displaysize=(99999,99999))
```

read more here: https://github.com/fonsp/Pluto.jl/pull/318#issuecomment-715251330


![image](https://user-images.githubusercontent.com/6933510/97051388-62599500-157f-11eb-80cc-a37e019fb7fd.png)

**Comment by fonsp** [comment]
Perhaps I haven't written this down but the plan is that Pluto will have built-in support for `Tables.jl` (the lightweight abstract supertype of DataFrames etc), with lazy loading, scrolling and more.

One of my big goals with Pluto is to be accessible to excel users

**Comment by lungben** [comment]
Thanks, sounds great!

**Comment by fonsp** [comment]
I have fixed #542, which means that we can now load in packages on the notebook process **like Tables.jl**

**Comment by fonsp** [comment]
I have started on building a Table viewer directly into Pluto:

https://github.com/fonsp/Pluto.jl/pull/646
